pkg/gather/collect/podcollector.go

```diff
@@ -0,0 +1,323 @@
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
+	oslices "github.com/scylladb/scylla-operator/pkg/helpers/slices"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	"github.com/scylladb/scylla-operator/pkg/pointer"
+	"github.com/scylladb/scylla-operator/pkg/scheme"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/klog/v2"
+)
+
+type PodRuntimeCollector struct {
+	Filter  func(pod *corev1.Pod) bool
+	Collect func(context.Context, *corev1.Pod, *ResourceInfo) error
+}
+
+type PodCollector struct {
+	restConfig     *rest.Config
+	corev1Client   corev1client.CoreV1Interface
+	resourceWriter *ResourceWriter
+	logsLimitBytes int64
+
+	podRuntimeCollectors []PodRuntimeCollector
+}
+
+func NewPodCollector(restConfig *rest.Config, corev1Client corev1client.CoreV1Interface, resourceWriter *ResourceWriter, logsLimitBytes int64) *PodCollector {
+	pc := &PodCollector{
+		restConfig:     restConfig,
+		corev1Client:   corev1Client,
+		resourceWriter: resourceWriter,
+		logsLimitBytes: logsLimitBytes,
+	}
+
+	pc.podRuntimeCollectors = []PodRuntimeCollector{
+		{
+			Collect: pc.collectPodLogs,
+		},
+		{
+			Filter:  controllerhelpers.IsScyllaContainerRunning,
+			Collect: pc.collectContainerCommandOutputFunc(naming.ScyllaContainerName, "nodetool-status.log", []string{"nodetool", "status"}),
+		},
+		{
+			Filter:  controllerhelpers.IsScyllaContainerRunning,
+			Collect: pc.collectContainerCommandOutputFunc(naming.ScyllaContainerName, "nodetool-gossipinfo.log", []string{"nodetool", "gossipinfo"}),
+		},
+		{
+			Filter:  controllerhelpers.IsScyllaContainerRunning,
+			Collect: pc.collectContainerCommandOutputFunc(naming.ScyllaContainerName, "df.log", []string{"df", "-h"}),
+		},
+		{
+			Filter:  controllerhelpers.IsScyllaContainerRunning,
+			Collect: pc.collectContainerCommandOutputFunc(naming.ScyllaContainerName, "io_properties.yaml", []string{"cat", "/etc/scylla.d/io_properties.yaml"}),
+		},
+		{
+			Filter: controllerhelpers.IsScyllaContainerRunning,
+			Collect: pc.collectContainerCommandOutputFunc(naming.ScyllaContainerName, "scylla-rlimits.log", []string{
+				"bash",
+				"-euEo",
+				"pipefail",
+				"-O",
+				"inherit_errexit",
+				"-c",
+				`prlimit --pid=$(pidof scylla)`}),
+		},
+		{
+			Filter:  controllerhelpers.IsNodeConfigPod,
+			Collect: pc.collectContainerCommandOutputFunc(naming.NodeConfigAppName, "kubelet-cpu_manager_state.log", []string{"cat", "/host/var/lib/kubelet/cpu_manager_state"}),
+		},
+	}
+
+	return pc
+}
+
+func retrieveContainerLogs(ctx context.Context, podClient corev1client.PodInterface, destinationPath string, podName string, logOptions *corev1.PodLogOptions) error {
```

Comment:
I think it will better reflect what this function does and also make it easier to distinguish from very similarly sounding `collectContainerLogs`.
```suggestion
func writeContainerLogsToFile(ctx context.Context, podClient corev1client.PodInterface, destinationPath string, podName string, logOptions *corev1.PodLogOptions) error {
```